src/JsonSchema/TypeFactory.php

```diff
@@ -169,10 +169,15 @@ private function addNullabilityToTypeDefinition(array $jsonSchema, Type $type, ?
         }
 
         if (\array_key_exists('$ref', $jsonSchema)) {
-            return [
-                'nullable' => true,
-                'anyOf' => [$jsonSchema],
-            ];
+            $typeDefinition = ['anyOf' => $jsonSchema];
+
+            if ($schema && Schema::VERSION_OPENAPI === $schema->getVersion()) {
```

Comment:
Please make sure you are in a JSON Schema context, not an OpenAPI one: nullable is not available for OpenAPI >= 3.1.